solutions/search/esql-full-text-filter-tutorial.md

```diff
@@ -288,125 +276,83 @@ Like the `term` query in Query DSL, this has zero flexibility and is case-sensit
 Complex searches often require combining multiple search criteria:
 
 ```esql
-POST /_query?format=txt
-{
-  "query": """
-    FROM cooking_blog METADATA _score
-    | WHERE rating >= 4.5
-      AND NOT category.keyword == "Dessert"
-      AND (title:"curry spicy" OR description:"curry spicy")
-    | SORT _score DESC
-    | KEEP title, author, rating, tags, description
-    | LIMIT 1000
-  """
-}
+FROM cooking_blog METADATA _score
+| WHERE rating >= 4.5  # Numerical filter
+    AND NOT category.keyword == "Dessert"  # Exclusion filter
+    AND (title:"curry spicy" OR description:"curry spicy")  # Full-text search in multiple fields
+| SORT _score DESC
+| KEEP title, author, rating, tags, description
+| LIMIT 1000
 ```
 
+### Combine relevance scoring with custom criteria
+
 For more complex relevance scoring with combined criteria, you can use the `EVAL` command to calculate custom scores:
 
 ```esql
-POST /_query?format=txt
-{
-  "query": """
-    FROM cooking_blog METADATA _score
-    | WHERE tags.keyword == "vegetarian" AND rating >= 4.5
-    | EVAL title_score = SCORE(match(title, "curry spicy")) * 2
-    | EVAL desc_score = SCORE(match(description, "curry spicy"))
-    | EVAL combined_score = title_score + desc_score
-    | EVAL category_boost = IF(category.keyword == "Main Course", 1.0, 0.0)
-    | EVAL date_boost = IF(date >= "now-1M/d", 0.5, 0.0)
-    | EVAL final_score = combined_score + category_boost + date_boost
-    | WHERE NOT category.keyword == "Dessert"
-    | WHERE final_score > 0
-    | SORT final_score DESC
-    | LIMIT 1000
-  """
-}
+FROM cooking_blog METADATA _score
+| EVAL tags_concat = MV_CONCAT(tags.keyword, ",")  # Convert multi-value field to string
+| WHERE tags_concat LIKE "*vegetarian*" AND rating >= 4.5  # Wildcard pattern matching
+| WHERE match(title, "curry spicy", {"boost": 2.0}) OR match(description, "curry spicy")
+| EVAL category_boost = CASE(category.keyword == "Main Course", 1.0, 0.0)  # Conditional boost
+| EVAL date_boost = CASE(DATE_DIFF("month", date, NOW()) <= 1, 0.5, 0.0)  # Boost recent content
+| EVAL custom_score = _score + category_boost + date_boost  # Combine scores
+| WHERE NOT category.keyword == "Dessert"
```

Comment:
Do you think we should move this closer to FROM, so it doesn't interfere with score calculation but is part of the filtering part?